libs/cli/src/main/java/org/elasticsearch/cli/Terminal.java

```diff
@@ -51,40 +52,87 @@ public enum Verbosity {
         VERBOSE /* printed only when cli is passed verbose option */
     }
 
+    private final Reader reader;
+
+    private final PrintWriter outWriter;
+
+    private final PrintWriter errWriter;
+
+    private final InputStream inStream;
+
+    private final OutputStream outStream;
+
     /** The current verbosity for the terminal, defaulting to {@link Verbosity#NORMAL}. */
     private Verbosity currentVerbosity = Verbosity.NORMAL;
 
-    /** The newline used when calling println. */
-    private final String lineSeparator;
-
-    protected Terminal(String lineSeparator) {
-        this.lineSeparator = lineSeparator;
+    protected Terminal(
```

Comment:
Either here or on the field definitions, I think we should document that the `inStream` if non-null is likely not independent of the `reader`, and the `outStream` if non-null is likely not independent of the `outWriter`, or at least there is no expectation of independence between the items in those pairs.

Reply:
I"ll take a pass at better documenting the entire class.